.github/workflows/test.yml

```diff
@@ -228,6 +228,100 @@ jobs:
           export PATH=$(pwd)/target/release:$PATH
           cargo test --manifest-path integration-tests/zombienet/Cargo.toml
 
+  build-runtimes:
+    needs: [ runtime-matrix ]
+    runs-on: ubuntu-latest
+    strategy:
+      # Ensure the other jobs are continue
+      fail-fast: false
+      matrix:
+        runtime: ${{ fromJSON(needs.runtime-matrix.outputs.runtime) }}
+    steps:
+      - name: Checkout sources
+        uses: actions/checkout@v4
+
+      - name: Set rust version via common env file
+        run: cat .github/env >> $GITHUB_ENV
+
+      - name: Install stable toolchain
+        run: |
+          rustup install $RUST_STABLE_VERSION
+          rustup default $RUST_STABLE_VERSION
+          rustup target add wasm32-unknown-unknown
+          rustup component add rust-src
+
+      - name: Fetch cache
+        uses: Swatinem/rust-cache@a95ba195448af2da9b00fb742d14ffaaf3c21f43 # v2.7.0
+        with:
+          shared-key: "fellowship-cache-build-runtimes-test"
+
+      - name: Build runtime ${{ matrix.runtime.name }}
+        run: |
+          PACKAGE_NAME=${{ matrix.runtime.package }}
+          RUNTIME_BLOB_NAME=$(echo $PACKAGE_NAME | sed 's/-/_/g').compact.compressed.wasm
+          RUNTIME_BLOB_PATH=./target/production/wbuild/$PACKAGE_NAME/$RUNTIME_BLOB_NAME
+          # Build wasm
+          echo "Building wasm RUNTIME_BLOB_PATH=$RUNTIME_BLOB_PATH"
+
+          # Dirty hack to find out if the `metadata-hash` feature exists for the given package.
+          # Should print "error: no bin target named `something-that-does-not-exist`." if the feature doesn't exist.
+          # Otherwise it should print "error: the package 'glutton-kusama-runtime' does not contain this feature: metadata-hash".
+          if ! cargo build -p ${{ matrix.runtime.package }} --features metadata-hash --bin something-that-does-not-exist 2>&1 | grep "does not contain this feature: metadata-hash"; then
```

Comment:
y u do dis, Sir?

can we do something like
```
cargo metadata --format-version=1 | jq '.packages[] | select(.name=="${{ matrix.runtime.package }}") | .features' | grep metadata-hash
```
that requires `jq` installed though, but maybe it's a small ask/could be preinstalled already?